### PR TITLE
Swap out unmaintained UUID library for Google one.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Had a look at what Kubernetes does; they have switched to the Google UUID library.

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by neiljerram — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
